assertj-error-prone/src/main/java/com/palantir/assertj/errorprone/AssertjInstanceOfAssert.java

```diff
@@ -0,0 +1,101 @@
+
+    private static final String DESCRIPTION =
+            "Prefer using AssertJ fluent comparisons over logic in an assertThat statement for better "
+                    + "failure output. assertThat(a instanceof b).isTrue() failures report 'expected true' where "
+                    + "assertThat(a).isInstanceOf(b) provides the expected and actual values.";
+
+    private static final Matcher<ExpressionTree> IS_TRUE = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .named("isTrue")
+            .withNoParameters();
+
+    private static final Matcher<ExpressionTree> IS_FALSE = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .named("isFalse")
+            .withNoParameters();
+
+    private static final Matcher<ExpressionTree> BOOLEAN_ASSERT = Matchers.anyOf(IS_TRUE, IS_FALSE);
+
+    private final AssertjSingleAssertMatcher matcher = AssertjSingleAssertMatcher.of(this::match);
+
+    @Override
+    public Optional<AssertjCheckerResult> matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        if (BOOLEAN_ASSERT.matches(tree, state)) {
+            return matcher.matches(tree, state);
+        }
+        return Optional.empty();
+    }
+
+    private Optional<AssertjCheckerResult> match(
+            AssertjSingleAssertMatcher.SingleAssertMatch match, VisitorState state) {
+        boolean negatedAssertion = IS_FALSE.matches(match.getCheck(), state);
+        if (!negatedAssertion && !IS_TRUE.matches(match.getCheck(), state)) {
+            return Optional.empty();
+        }
+        ExpressionTree target = match.getAssertThat().getArguments().get(0);
+        ExpressionTree instanceOfExpression = target;
+
+        if (target instanceof UnaryTree) {
+            UnaryTree unaryTree = (UnaryTree) target;
+            if (unaryTree.getExpression() instanceof ParenthesizedTree) {
+                negatedAssertion = !negatedAssertion;
+                ParenthesizedTree parenthesizedTree = (ParenthesizedTree) unaryTree.getExpression();
+                instanceOfExpression = parenthesizedTree.getExpression();
+            }
+        }
+        if (!(instanceOfExpression instanceof InstanceOfTree)) {
+            return Optional.empty();
+        }
+        InstanceOfTree instanceOfTree = (InstanceOfTree) instanceOfExpression;
+
+        ExpressionTree actual = instanceOfTree.getExpression();
+        Tree expected = instanceOfTree.getType();
+
+        SuggestedFix fix = SuggestedFix.builder()
+                .replace(
+                        state.getEndPosition(
+                                ((MemberSelectTree) match.getCheck().getMethodSelect()).getExpression()),
+                        state.getEndPosition(match.getCheck()),
+                        String.format(
+                                negatedAssertion ? ".isNotInstanceOf(%s.class)" : ".isInstanceOf(%s.class)", expected))
```

Comment:
is there a better way to do the `.class` part here?

Reply:
nah, this is reasonable. I'd recommend pulling the `expected` type info using `getSourceForNode` though